openfe/setup/methods/rbfe_methods.py

```diff
@@ -0,0 +1,257 @@
+# This code is part of OpenFE and is licensed under the MIT license.
+# For details, see https://github.com/OpenFreeEnergy/openfe
+"""Contains RBFE methods
+
+This module implements.....
+
+"""
+from __future__ import annotations
+
+from openfe.setup import LigandAtomMapping, LigandMolecule
+from openfe.setup.methods import FEMethod
+from typing import Dict, Union
+from pydantic import BaseModel, validator
+from openff.units import unit
+
+# define a timestep
+# this isn't convertible to time (e.g ps) and is used to not confuse these two
+# definitions of "duration" within a simulation
+unit.define('timestep = [timestep] = _ = timesteps')
+
+
+class ForcefieldSettings(BaseModel):
+    """Settings describing the forcefield to use for each component
+
+    Attributes
+    ----------
+    forcefield
+      a mapping of each components name to the xml forcefield to apply
+    nonbonded_electrostatics
+      which nonbonded electrostatics method to use, currently only 'PME'
+      allowed
+    nonbonded_cutoff
+      range of nonbonded forces
+    constraints
+      which bonds to apply constraints to
+    rigid_water
+      whether apply rigid constraints to water molecules, default True
+    """
+    class Config:
+        arbitrary_types_allowed = True
+
+    # mapping of component name to forcefield path
+    forcefield: Dict[str, str]
+    nonbonded_electostatics = 'PME'
+    nonbonded_cutoff = 0.9 * unit.nanometer
+    constraints = 'HBonds'
+    rigid_water = True
```

Comment:
Will add hydrogenMass.

One thing these `Settings` classes also allow/force is a degree of separation between the user settings and the toolkit.  This will occasionally be frustrating (e.g. when I want to change `hydrogenMass` and we forgot to expose it) but it also means that the `Protocol` object can also smooth out any API breaks that happen (i.e. put in two paths around an API break).